<!-- <?xml version="1.0"?> -->
<launch>
    <arg name="task" default="Rbolts"/>
    <arg name="subtask" default="action"/>

    <param name="task_param" type="string" value="$(arg task)"/>
    <param name="subtask_param" type="string" value="$(arg subtask)"/>
    

    <!-- <node pkg="tf" type="static_transform_publisher" name="gmm_base_frame_ur10_1" args="0.0 0.0 0.0 0.0 0.0 0.7071068 0.70710680 /ur10_1_base_link /ur10_1_base_link_gmm 2"/>    
    <node pkg="tf" type="static_transform_publisher" name="gmm_base_frame_ur10_2" args="0.0 0.0 0.0 0.0 0.0 1.0 0.0 /ur10_2_base_link /ur10_2_base_link_gmm 2"/>
    <node pkg="tf" type="static_transform_publisher" name="gmm_flange_frame_ur10_1" args="0.0 0.0 0.0 0.7071068 0.0 0.0 0.7071068 /ur10_1_flange /ur10_1_flange_gmm 2"/>    
    <node pkg="tf" type="static_transform_publisher" name="gmm_flange_frame_ur10_2" args="0.0 0.0 0.0 0.7071068 0.0 0.0 0.7071068 /ur10_2_flange /ur10_2_flange_gmm 2"/> -->

    <node pkg="tf2_ros" type="static_transform_publisher" name="gmm_base_frame_ur10_1" args="0.0 0.0 0.0 0.0 0.0 0.7071068 0.70710680 /ur10_1_base_link /ur10_1_base_link_gmm"/>    
    <node pkg="tf2_ros" type="static_transform_publisher" name="gmm_base_frame_ur10_2" args="0.0 0.0 0.0 0.0 0.0 1.0 0.0 /ur10_2_base_link /ur10_2_base_link_gmm"/>
    <node pkg="tf2_ros" type="static_transform_publisher" name="gmm_flange_frame_ur10_1" args="0.0 0.0 0.0 0.7071068 0.0 0.0 0.7071068 /ur10_1_flange /ur10_1_flange_gmm"/>    
    <node pkg="tf2_ros" type="static_transform_publisher" name="gmm_flange_frame_ur10_2" args="0.0 0.0 0.0 0.7071068 0.0 0.0 0.7071068 /ur10_2_flange /ur10_2_flange_gmm"/>


    <!-- <node name="demons_entry" pkg="data_handle" type="demons_entry_node" output="screen"/> -->
    <node name="tp_gmm" pkg="tp_gmm" type="tp-gmm.py" output="screen"/>

    <!-- GMM -->
    <!-- <node name="gaussian_mixture_model_trajectory_converter" pkg="gaussian_mixture_model" type="gaussian_mixture_model_trajectory_converter" output="screen"/> -->
    <!-- <node name="gaussian_mixture_model_node" pkg="gaussian_mixture_model" type="gaussian_mixture_model_node" output="screen"/> -->
    <node name="gmm_rviz_converter_node" pkg="tp_gmm" type="gmm_rviz_converter_node" output="screen"/>
    <node name="gmm_moveit_node" pkg="tp_gmm" type="gmm_moveit_node" output="screen"/>    
    <!--/ GMM -->

    <!-- Execution -->
    <!-- <node name="execute_trajectory" pkg="data_handle" type="execute_trajectory_node" output="screen"/>
    <node name="move_trajectory" pkg="data_handle" type="move_traj.py" output="screen"/> -->
    <!--/ Execution -->

</launch>